android/src/main/java/com/tailscale/ipn/ui/view/MainView.kt

```diff
@@ -526,154 +534,117 @@ fun PeerList(
-
   val focusManager = LocalFocusManager.current
   var isFocussed by remember { mutableStateOf(false) }
-
   var isListFocussed by remember { mutableStateOf(false) }
-
   val expandedPeer = viewModel.expandedMenuPeer.collectAsState()
   val localClipboardManager = LocalClipboardManager.current
-
   val enableSearch = !isAndroidTV()
 
-  if (enableSearch) {
-    Box(modifier = Modifier.fillMaxWidth().background(color = MaterialTheme.colorScheme.surface)) {
-      OutlinedTextField(
-          modifier =
-              Modifier.fillMaxWidth()
-                  .padding(start = 16.dp, end = 16.dp, top = 16.dp, bottom = 0.dp)
-                  .onFocusChanged { isFocussed = it.isFocused },
-          singleLine = true,
-          shape = MaterialTheme.shapes.extraLarge,
-          colors = MaterialTheme.colorScheme.searchBarColors,
-          leadingIcon = {
-            Icon(imageVector = Icons.Outlined.Search, contentDescription = "search")
-          },
-          trailingIcon = {
-            if (isFocussed) {
-              IconButton(
-                  onClick = {
-                    focusManager.clearFocus()
-                    onSearch("")
-                  }) {
-                    Icon(
-                        imageVector =
-                            if (searchTermStr.isEmpty()) Icons.Outlined.Close
-                            else Icons.Outlined.Clear,
-                        contentDescription = "clear search",
-                        tint = MaterialTheme.colorScheme.onSurfaceVariant)
-                  }
-            }
-          },
-          placeholder = {
-            Text(
-                text = stringResource(id = R.string.search),
-                style = MaterialTheme.typography.bodyLarge,
-                maxLines = 1)
-          },
-          value = searchTermStr,
-          onValueChange = { onSearch(it) })
-    }
-  }
+  Column(modifier = Modifier.fillMaxSize()) {
+    SearchWithDynamicSuggestions(viewModel, onSearch)
 
-  LazyColumn(
-      modifier =
-          Modifier.fillMaxSize()
-              .onFocusChanged { isListFocussed = it.isFocused }
-              .background(color = MaterialTheme.colorScheme.surface)) {
-        if (showNoResults) {
-          item {
-            Spacer(
-                Modifier.height(16.dp)
-                    .fillMaxSize()
-                    .focusable(false)
-                    .background(color = MaterialTheme.colorScheme.surface))
-
-            Lists.LargeTitle(
-                stringResource(id = R.string.no_results),
-                bottomPadding = 8.dp,
-                style = MaterialTheme.typography.bodyMedium,
-                fontWeight = FontWeight.Light)
-          }
-        }
+    Spacer(modifier = Modifier.height(if (showNoResults) 0.dp else 8.dp))
 
-        var first = true
-        peerList.forEach { peerSet ->
-          if (!first) {
-            item(key = "user_divider_${peerSet.user?.ID ?: 0L}") { Lists.ItemDivider() }
+    // Peers display
```

Comment:
This isn't wrapped in the enableSearch flag... Now, this might actually work better on AndroidTV now!  but I tried it on a 720p simulator and you can't see the search results - there's just not enough vertical space.  I'm OK with just hiding the search bar on TV for now but ideally, we should make it work.  The hand-rolled search bar we were using before had several issues...

Reply:
yeah, unfortunately the keyboard just takes up so much of the screen on 720p. I've readded that enableSearch check, I'll punt making the search bar work for TV for now.

Reply:
I added the check back - the keyboard takes up so much space in 720p. Agreed we should have search on TV, but this might require a separate design for TV.